### PR TITLE
improve worst case image build time

### PR DESCRIPTION
**Pull request by jakecoffman**
In #5356 we switched away from a PPA since it wasn't keeping up with new versions of Ruby. Ruby-install was good but building from source is slow, and it seems like we break the cache more often than we care to.

This approach is to copy Ruby et al from the official Ruby Docker image which should stay up-to-date. This allows us to skip the build to save time, and we can stay on our current version of Ubuntu by copying out the binaries. This also saves ~100MB from somewhere, the `bundler` image is now around 640MB for me. 

I had to move the `bundle install` out since it didn't seem to like to be run on a different version of Ubuntu? I'm not sure, but even with less parallelism this is very fast locally. Submitting as Draft to test against CI, check the speed, and get feedback.

### CODE REVIEW

**Comment by jakecoffman** [comment]
An uncached build time of 1:30-1:40 for most ecosystems!

**Comment by jakecoffman** [comment]
@pavera Do you think we could do something similar with Python? Might be hacky since we depend on pyenv.

**Review by jeffwidman — APPROVED** [review]
Clever.

Feels a little risky to me, but I assume the official Ruby binary is statically linked?

I know one thing we'd also discussed before was the idea of using the pre-compiled binary that the GitHub Actions team maintains... this seems more straightforward though...

**Review by deivid-rodriguez — APPROVED** [review]
This is really awesome! I had the same idea for Python and did give it a try but couldn't make it work.

File: go_modules/spec/dependabot/go_modules/update_checker/latest_version_finder_spec.rb [hunk 1]
```diff
@@ -157,7 +157,8 @@ module foobar
       let(:dependency_name) { "golang.org/x/net" }
 
       it "picks the latest version" do
-        expect(finder.latest_version).to eq(Dependabot::GoModules::Version.new("0.8.0"))
+        # This makes actual requests so we should use a VCR cassette
+        expect(finder.latest_version).to eq(Dependabot::GoModules::Version.new("0.9.0"))
```
[the remark this answers is not in the document]
Comment (jakecoffman) [reply go_modules/spec/dependabot/go_modules/update_checker/latest_version_finder_spec.rb]: Unrelated fixed by https://github.com/dependabot/dependabot-core/pull/7016

**Review by jeffwidman — APPROVED** [review]

**Pull request merged**